pkg/apis/kops/validation/validation.go

```diff
@@ -1382,6 +1382,14 @@ func validateEtcdClusterSpec(spec kops.EtcdClusterSpec, c *kops.Cluster, fieldPa
 		allErrs = append(allErrs, validateEtcdMemberSpec(m, fieldPath.Child("etcdMembers").Index(i))...)
 	}
 
+	if spec.Manager != nil && spec.Manager.NetworkCIDR != nil {
+		networkCIDRPath := fieldPath.Child("manager").Child("networkCIDR")
+		if c.Spec.CloudProvider.Openstack == nil {
```

Comment:
if c.Spec.GetCloudProvider() != api.CloudProviderOpenstack

perhaps better way to check the cloud, that is what is used in most of the places in kops